linode/manager
# change: [M3-8058] - Add tags drawer for Kubernetes Cluster tag overflow

## Description 📝
Changes the tags list in `KubeSummaryPanel` from an expanding list to a single line with an overflow drawer.

This PR also includes a refactor to `TagCell` that bundles the `TagDrawer` instead of requiring other components to include their own.

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Tag list takes up many lines when multiple tags are added ![Screenshot 2024-07-08 at 6 05 44 PM](https://github.com/linode/manager/assets/122488130/beefa601-6a4a-448d-87ba-74a2d2a4ad28) | Displays "..." button when overflowing ![Screenshot 2024-07-08 at 6 05 20 PM](https://github.com/linode/manager/assets/122488130/2af2c79a-4c65-4cd4-a9ce-a4f1b3deb6d5) Tags drawer displays all tags  ![Screenshot 2024-07-08 at 6 06 02 PM](https://github.com/linode/manager/assets/122488130/bf3707bf-515b-4826-87c0-50a363ec7fae) |

## How to test 🧪

### Verification steps
- Navigate to the details page of a Kubernetes cluster
- Add many tags and observe that instead of wrapping into multiple lines, the overflow is hidden and a button appears to display the tag drawer.

## As an Author I have considered 🤔
